### ISSUE TITLE
Audio stops when locking the device

### ISSUE THREAD

**Issue by YoussefAK** [0xced/XCDYouTubeKit#256]
Hi, I have implemented your "hack" that allows background mode, it works when I press the home button, but when I lock the screen, it just update my playback status to paused. Is there something I am missing?
Thanks for your help

**Comment by JJGEight** [comment]
I've had the same problem. Using the home button and then pressing the power button to lock the device will work. However if I am in the app and then press the lock button, the audio will stop.

I'm guessing the problem is caused because `UIApplicationWillResignActiveNotification` is never called while the app is still open and you press the power button to lock the screen. I've looked at two methods to try and fix it. The first being was adding the following line to the load method in MPMoviePlayerController+BackgroundPlayback.

`[defaultCenter addObserver:self selector:@selector(backgroundPlayback_applicationWillResignActive:) name:UIApplicationDidEnterBackgroundNotification object:nil];`

This didn't work and I'm assuming this is because it's too late to run the code which strips the video layer out of AVPlayer. (if only there was a `UIApplicationWillEnterBackgroundNotification` notification).

The second attempt I took was following [Playing media while in the background using AV Foundation on iOS](https://developer.apple.com/library/ios/qa/qa1668/_index.html). However all this achieved was my own version of MPMoviePlayerController+BackgroundPlayback which really didn't achieve much.

I'm not really too sure what else can be done. Does anyone else have any thoughts?

**Comment by letungit90** [comment]
Had same problem

**Comment by TheRyanHickman** [comment]
Has anyone solved this, as I am also having the same issue.

**Comment by brevansio** [comment]
I don't use MPPlayer or the XCD player (XCDYouTubeClient for URL + AVPlayer), but I did get pauses a while back, so it might be helpful. In my case it had to do with AVPlayerLayer getting set/unset. (see WWDC2016 Session 503)

I fixed my issue by monitoring `UIApplicationDidEnterBackgroundNotification` and `UIApplicationDidBecomeActiveNotification` and calling play if needed. Pre-iOS 10 this did cause a 0.5~1s pause, but as of iOS10, there is no interruption.

These two notifications always get called when pressing Home or when locking the phone.
